content/docs/concepts/traffic-management/index.md

```diff
@@ -678,6 +559,146 @@ spec:
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: ratings
+spec:
+  hosts:
+  - ratings
+  http:
+  - match:
+      sourceLabels:
+        app: reviews
+    ...
+{{< /text >}}
+
+The value of `sourceLabels` depends on the implementation of the service.
+In Kubernetes, for example, it would probably be the same labels that are used
+in the pod selector of the corresponding Kubernetes service.
+
+_2. Restrict to specific versions of the caller_. For example, the following
+rule refines the previous example to only apply to calls from version "v2"
+of the *reviews* service.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: ratings
+spec:
+  hosts:
+  - ratings
+  http:
+  - match:
+    - sourceLabels:
+        app: reviews
+        version: v2
+    ...
+{{< /text >}}
+
+_3. Select rule based on HTTP headers_. For example, the following rule will
+only apply to an incoming request if it includes a "cookie" header that
+contains the substring "user=jason".
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: reviews
+spec:
+  hosts:
+    - reviews
+  http:
+  - match:
+    - headers:
+        cookie:
+          regex: "^(.*?;)?(user=jason)(;.*)?$"
+    ...
+{{< /text >}}
+
+If more than one header is provided, then all of the
+corresponding headers must match for the rule to apply.
+
+_4. Select rule based on request URI_. For example, the following rule will
+only apply to a request if the URI path starts with `/api/v1`.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: productpage
+spec:
+  hosts:
+    - productpage
+  http:
+  - match:
+    - uri:
+        prefix: /api/v1
+    ...
+{{< /text >}}
+
+#### Multiple match conditions
+
+Multiple match criteria can be set simultaneously. In such a case, AND or OR
+semantics apply, depending on the nesting.
+
+If multiple criteria are nested in a single match clause, then the conditions
+are ANDed. For example, the following rule only applies if the source of the
+request is "reviews:v2" AND the "cookie" header containing "user=jason" is
+present.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: ratings
+spec:
+  hosts:
+  - ratings
+  http:
+  - match:
+    - sourceLabels:
+        app: reviews
+        version: v2
+      headers:
+        cookie:
+          regex: "^(.*?;)?(user=jason)(;.*)?$"
+    ...
+{{< /text >}}
+
+If instead, the criteria appear in separate match clauses, then only one
+of the conditions must apply (OR semantics):
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: ratings
+spec:
+  hosts:
+  - ratings
+  http:
+  - match:
+    - sourceLabels:
+        app: reviews
+        version: v2
+    - headers:
+        cookie:
+          regex: "^(.*?;)?(user=jason)(;.*)?$"
```

Comment:
May be worth stating again that this will apply to clients with labels .. OR requests with headers ...